screens/DoubleSearch.js

```diff
@@ -9,103 +9,228 @@ import { buildingData } from "../constants/buildingData";
 
-    const goToBackToMoreDetails = () => {
+    /**
+     * Description: Method to go back to the previous screen.
+     * Using Stack navigator.
+     */
+    const goBack = () => {
         props.navigation.goBack();
     };
-    const getDirectionScreen = () => {
-        props.navigation.navigate("MapDirections");
+    /**
+     * Description: This method will navigate between the DoubleSearch screen to the PreviewDirection screen.
+     * Particularity: 
+     * 1. We use Stack navigator
+     * 2. Error Handling to prevent unwanted behavior in rendering of map in subsequent screens (PreviewDirection &
+     * Direction).
+     *      - The same address for origin and destination is not accepted
+     *      - If the Current Direction is wanted, fetch the proper coordinates
+     *      - If there's no destination, refuse search
+     *      - A value selected beyond the search Items displayed by the dropdown is not accepted.
+     * 
+     * A.U
+     */
+    const goToPreviewDirectionScreen = () => {
+        if (to.name == from.name) {
+            return alert("Origin and destination are the same. Please try Again.");
+        }
+        else if (coordinatesFrom && coordinatesTo) {
+            props.navigation.navigate("PreviewDirections", { From: coordinatesFrom, To: coordinatesTo });
+        }
+        else if (from.name == "Current Location" && currentLocationCoords.latitude && currentLocationCoords.longitude) {
+            props.navigation.navigate("PreviewDirections", { From: currentLocationCoords, To: coordinatesTo });
+        }
+        //TODO: Refer To A)
+        else if (currentLocationCoords.latitude && currentLocationCoords.longitude) {
+            props.navigation.navigate("PreviewDirections", { From: currentLocationCoords, To: coordinatesTo });
+        }
+        else if (from.name == "Current Location" && !currentLocationCoords) {
+            return alert("Error: Are location services on?");
+        }
+        else {
+            return alert("The destination field is missing or you typed an invalid location. Please try again.");
+        }
     };
+    
+    const destinationName =  props.navigation.getParam("destinationName", "Destination");
 
-    const selectDestinationName = () => {
-        return props.navigation.getParam("destinationName", "null");
+    /**
+     * Algorithm to find the coordinates of a given building name.
+     * returns longitude and latitude
+     * TODO: B) Another algorithm or extend this one to take in consideration classrooms.
+     * A.U
+     * @param {*} name 
+     */
+    const getCoordinates = (name) => {
+
+        var list = buildingData();
+        for (var key in list) {
+            if (list[key].name.includes(name)) {
+                return list[key].coordinates;
+            }
+        }
+        return null;
+    };
+
+    const getCurrentLocation = () => {
```

Comment:
We would have to modify this method a bit as well since we are setting a state inside of it. If we had Redux, no problem, but as for now, states are local to their functions.

Reply:
CurrentBuildingLocation also has similar function to retrieve the current Location. Please check with that file also

Reply:
Yes I do know. It's the most straightforward way to get the CurrentLocation. I also use a similar function to zoom mapview to the user current location. The thing is, this is a React Native function, only what is within the body will change from components to components. It would have been perfect if we had Redux so that this function updates a state in the Redux Store. _**To investigate**_